src/main/kotlin/org/rust/cargo/toolchain/tools/CargoConfig.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.toolchain.tools
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.core.JacksonException
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.dataformat.toml.TomlMapper
+import com.intellij.openapi.project.Project
+import org.jetbrains.annotations.VisibleForTesting
+import org.rust.cargo.toolchain.RsToolchainBase
+import org.rust.openapiext.RsDeserializationException
+import org.rust.openapiext.RsProcessExecutionOrDeserializationException
+import org.rust.openapiext.execute
+import org.rust.stdext.RsResult
+import org.rust.stdext.RsResult.Err
+import org.rust.stdext.RsResult.Ok
+import org.rust.stdext.unwrapOrElse
+import java.nio.file.Path
+
+fun Cargo.config(owner: Project, projectDirectory: Path?): CargoConfig =
+    CargoConfig(this.toolchain, owner, projectDirectory)
+
+class CargoConfig(
+    toolchain: RsToolchainBase,
+    private val owner: Project,
+    private val projectDirectory: Path?
+) : RsTool("cargo", toolchain) {
+    /**
+     * Execute `cargo config get <path>` to and parse output as Jackson Tree ([JsonNode]).
+     * Use [JsonNode.at] to get properties by path (in `/foo/bar` format)
+     */
+    @VisibleForTesting
+    fun getConfig(path: String): RsResult<JsonNode, RsProcessExecutionOrDeserializationException> {
```

Comment:
The current implementation makes us execute external process each time when we need something from cargo config but it's not what we want. We want to execute it once during updating project model, keep its result in some class (I suppose it should be named `CargoConfig`)